CHANGELOG_v4_draft.md

```diff
@@ -0,0 +1,117 @@
+- The field `GraphQLRequestContext.logger` is now readonly.
+- The fields `GraphQLRequestContext.schemaHash` and `GraphQLRequestContext.debug` have been removed.
+- The type `GraphQLServiceContext` has been renamed to `GraphQLServerContext`, and the fields `schemaHash`, `persistedQueries`, and `serverlessFramework` have been removed; the latter has been semi-replaced by `startedInBackground`.
+- The `http` field on the `GraphQLRequest` object (available to plugins as `requestContext.request` and as an argument to `server.executeOperation`) is no longer based on the Fetch API's `Request` object. It no longer contains an URL path, and its `headers` field is a `Map` rather than a `Headers` object.
+- The structure of the `GraphQLResponse` object (available to plugins as `requestContext.response` and as the return value from `server.executeOperation`) has [changed in several ways](https://www.apollographql.com/docs/apollo-server/migration/#graphqlresponse).
+- The `plugins` constructor argument does not take factory functions.
+- `requestDidStart` hooks are called in parallel rather than in series.
+- A few changes have been made which may affect [custom `gateway` and `GraphQLDataSource` implementations](https://www.apollographql.com/docs/apollo-server/migration/#custom-gateway-and-graphqldatasource-implementations).
+
+
+#### Changes to defaults
+
+- CSRF prevention is on by default.
+- HTTP batching is disabled by default.
+- The default in-memory cache is bounded.
+- The local landing page defaults to the *embedded* Apollo Sandbox; this provides a user interface for executing GraphQL operations which doesn't require any additional CORS configuration.
+- The usage reporting and inline trace plugins mask errors in their reports by default: error messages are replaced with `<masked>` and error extensions are replaced with a single extension `maskedBy`. This can be configured with the `sendErrors` option to `ApolloServerPluginUsageReporting` and the `includeErrors` option to `ApolloServerPluginInlineTrace`. The `rewriteError` option to these plugins has been removed; its functionality is subsumed by the new options.
+
+#### TypeScript-specific changes
+
+- The TypeScript types for the `validationRules` constructor option are more accurate.
+- We now use the `@apollo/utils.fetcher` package to define the shape of the Fetch API, instead of `apollo-server-env`. This package only supports argument structures that are likely to be compatible across implementations of the Fetch API.
+- The `CacheScope`, `CacheHint`, `CacheAnnotation`, `CachePolicy`, and `ResolveInfoCacheControl` types are now exported from the `@apollo/cache-control-types` package. `CacheScope` is now a pure TypeScript type rather than an enum.
+- The type for `ApolloServer`'s constructor options argument is now `ApolloServerOptions`, not `Config` or `ApolloServerExpressConfig`.
+- Some other types have been renamed or removed; see the migration guide for details.
+
+### New features
```

Comment:
The only other thing I'd suggest is moving these new features up, maybe below "New package structure"? Folks love to read about new stuff and they might not make it through all the other changes to the bottom of this file

Reply:
Yeah, I do feel like breaking changes coming first makes sense though? The fact that this release is 95% breaking change does certainly hide the new features though.